Dockerfile

```diff
@@ -1,4 +1,4 @@
-FROM ubuntu:20.04
+FROM ubuntu:20.04.3
```

Comment:
please do not change this.

Reply:
because I met a problem, CI failed in installing llvm12 with error "**libclang-cpp12 (>= 1:12.0.1~++20210802051739+fed41342a82f) but 1:12.0.0-3ubuntu1~20.04.3 is to be installed" , I found those necessary dependencies "Package "llvm-toolchain-12" was removed from ubuntu 20.04**
https://www.ubuntuupdates.org/package/core/focal/universe/proposed/llvm-toolchain-12

Reply:
This is most likely not related to your change.

Reply:
yes, not related to my change

Reply:
but if we want to run the new test for uefi on qemu,  we need to rebuild a CI docker, so I want to find a method to fix this.

Reply:
this is broken in the llvm apt repo I think and could be temporary.

Reply:
@galak  I thinks it's the same issue in https://github.com/zephyrproject-rtos/docker-image/pull/70/commits/4e9fc7ecef5275e30ad4fa70f98adfa614ec4b7f.
if we used llvm.sh script, it will install llvm from repo https://apt.llvm.org/focal llvm-toolchain-focal-12/main, it will use the latest version 1:12.0.1, using libclang-cpp12 as an example
Get:21 https://apt.llvm.org/focal llvm-toolchain-focal-12/main amd64 libclang-cpp12 amd64 1:12.0.1~++20210802051739+fed41342a82f-1~exp1~20210802152501.132 [9707 kB]

but if we install llvm12 directly in ubuntu 20.04, it will get llvm from http://archive.ubuntu.com/ubuntu focal-updates/universe and it only includes 1:12.0.0 version.
Get:4 http://archive.ubuntu.com/ubuntu focal-updates/universe amd64 libclang-cpp12 amd64 1:12.0.0-3ubuntu1~20.04.3 [9,661 kB]

sometimes it will mix up, i.e, it get clang-12 from https://apt.llvm.org/focal llvm-toolchain-focal-12/main, it's version is 1:12.0.1, but it got libclang-cpp12 from http://archive.ubuntu.com/ubuntu focal-updates/universe, it's version is 1:12.0.0, then it will report the error above.

the CI passed of this time is that this step "add-apt-repository "${REPO_NAME}"" in llvm.sh script hasn't been executed, so it got llvm from ubuntu directly and didn't report any errors.
 
this is command I copied from CI pipeline, it lacked "add-apt-repository"
RUN wget ${WGET_ARGS} -O - https://apt.llvm.org/llvm-snapshot.gpg.key | apt-key add - && 	apt-get update && 	apt-get install -y clang-$LLVM_VERSION lldb-$LLVM_VERSION lld-$LLVM_VERSION clangd-$LLVM_VERSION 

**so I think we should install llvm directly, don't use llvm.sh script.**

Reply:
@chen-png is there some issue with the change I made in the commit you referenced where I moved away from use of llvm.sh?

Reply:
no, it worked well to install llvm12, I tested it on my machine.

Reply:
@galak I think it's ok to use your change in 4e9fc7e to install llvm.